include/net/phy.h

```diff
@@ -0,0 +1,231 @@
+/**
+ * @file
+ *
+ * @brief Public APIs for Ethernet PHY drivers.
+ */
+
+/*
+ * Copyright (c) 2021 IP-Logix Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PHY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PHY_H_
+
+/**
+ * @brief Ethernet PHY Interface
+ * @defgroup ethernet_phy Ethernet PHY Interface
+ * @ingroup networking
+ * @{
+ */
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief Ethernet link speeds. */
+enum phy_link_speed {
+	/** 10Base-T Half-Duplex */
+	LINK_HALF_10BASE_T		= BIT(0),
+	/** 10Base-T Full-Duplex */
+	LINK_FULL_10BASE_T		= BIT(1),
+	/** 100Base-T Half-Duplex */
+	LINK_HALF_100BASE_T		= BIT(2),
+	/** 100Base-T Full-Duplex */
+	LINK_FULL_100BASE_T		= BIT(3),
+};
+
+#define PHY_LINK_IS_FULL_DUPLEX(x)		(x & (BIT(1) | BIT(3)))
+#define PHY_LINK_IS_SPEED_100M(x)		(x & (BIT(2) | BIT(3)))
+
+/** @brief Link state */
+struct phy_link_state {
+	/** Link speed */
+	enum phy_link_speed speed;
+	/** When true the link is active and connected */
+	bool is_up;
+};
+
+/**
+ * @typedef phy_callback_t
+ * @brief Define the callback function signature for
+ * `phy_link_callback_set()` function.
+ *
+ * @param dev       PHY device structure
+ * @param state     Pointer to link_state structure.
+ * @param user_data Pointer to data specified by user
+ */
+typedef void (*phy_callback_t)(const struct device *dev,
+				      struct phy_link_state *state,
+				      void *user_data);
+
+/**
+ * @cond INTERNAL_HIDDEN
+ *
+ * These are for internal use only, so skip these in
+ * public documentation.
+ */
+__subsystem struct ethphy_driver_api {
```

Comment:
Same comment to subsystem and syscalls, these sounds unnecessary unless you have a good use case in mind.